docs/api/v1.md

```diff
@@ -425,37 +426,31 @@ Todo....
 ### SourceValue
 ```
 {
-  sTypeIndex {integer}: Index into SourceTypeDefinition array of the type that this variable is.
-  data {boolean | integer | double | null}: Only used if sTypeIndex points to a type of `kind == "primitive"`. Contains the primitive data.
-  target {integer}: Only used if sTypeIndex points to a type of `kind == "pointer"`. Contains the index into the SourceType array that this points to.
-  fields { name {string}: integer}: Only used if sTypeIndex points to a type of `kind == "struct"`. Map of indexes into the SourceType array that contain definitions for each field of a given name. 
+  sTypeIndex {integer}: Index into SourceType array of the type that this value is.
+  data If `kind == "primitive"` {boolean | integer | double | null}: Contains the primitive data.
+       If `kind == "pointer"` {integer}: Contains the index into the SourceType array that this points to.
+	   If `kind == "struct"` {name {string}: integer}: Map of indexes into the SourceType array that contain definitions for each field of a given name. 
 }
 ```
 
-### SourceTypeDefinition
+### SourceType
 ```
 {
   kind {string}: Type of variable. Can be "primitive" | "pointer" | "struct".
-  primitive {SourcePrimitiveDefinition}: Only used if `kind == "primitive"`. Defines primitive information
-  sTypeIndex {integer}: Only used if `kind == "pointer"`. Index into array of SourceTypeDefinitions array of the type that this pointer points to.
-  name {string}: Only used if `kind == "struct"`. Name of the struct as used in source code.
-  fields {SourceStructFieldDefinition[]}: Only used if `kind == "struct"`. Array of fields on this struct.
+  sTypeIndex {integer}: Only used if `kind == "pointer"`. Index into array of SourceType array of the type that this pointer points to.
+  name {string}: If `kind == "primitive"` can be "void" | "bool" | "int" | "uint" | "float". 
```

Comment:
Should put `name` and `fields` in a struct-specific variant; then base type, signedness, and size in a primitive-specific variant.